ui/src/components/CustomControl/CustomControlMockForTest.ts

```diff
@@ -0,0 +1,76 @@
+import { Mode } from '../../constants/modes';
+import { AcceptableFormValueOrNullish } from '../../types/components/shareableTypes';
+import { GlobalConfig } from '../../types/globalConfig/globalConfig';
+import { UtilControlWrapper } from '../BaseFormView/BaseFormTypes';
+
+export class CustomControlMockForTest {
+    globalConfig: GlobalConfig;
+
+    el: Element;
+
+    data: { mode: Mode; serviceName: string; value: AcceptableFormValueOrNullish };
+
+    util: UtilControlWrapper;
+
+    setValue: (newValue: AcceptableFormValueOrNullish) => void;
+
+    /**
+     *
+     * @constructor
+     * @param {object} globalConfig - Global configuration.
+     * @param {element} el - The element of the custom row.
+     * @param {object} data - Mode, serviceName, and value.
+     * @param {function} setValue - set value of the custom field.
+     * @param {object} util - The utility object.
+     */
+    constructor(
+        globalConfig: GlobalConfig,
+        el: Element,
+        data: { mode: Mode; serviceName: string; value: AcceptableFormValueOrNullish },
+        setValue: (newValue: AcceptableFormValueOrNullish) => void,
+        util: UtilControlWrapper
+    ) {
+        this.globalConfig = globalConfig;
+        this.el = el;
+        this.data = data;
+        this.util = util;
+        this.setValue = setValue;
+        this.onSelectOptionChange = this.onSelectOptionChange.bind(this);
+    }
+
+    onSelectOptionChange(event?: { target: EventTarget | null } | null) {
+        this.setValue((event?.target as HTMLSelectElement)?.value);
+    }
+
+    /* eslint-disable class-methods-use-this */
+    validation(_field: string, value: string) {
+        // Validation logic for value. Return the error message if failed.
+        if (value === 'input_two') {
+            return 'Wrong value selected.';
+        }
+
+        return undefined;
+    }
+
+    render() {
+        const options = ['input_default', 'input_one', 'input_two', 'input_three'].map(
+            (value) =>
+                `<option value="${value}" ${
+                    value === this?.data?.value && 'selected'
```

Comment:
if `value !== this?.data?.value` there would be 
```tsx
<option value="val1" false>${value}</option>
``` 
maybe to omit false?

Reply:
changed it into this one
`value === this?.data?.value ? 'selected' : ''`